WordPress/src/main/java/org/wordpress/android/ui/blaze/ui/blazewebview/BlazeWebViewFragment.kt

```diff
@@ -0,0 +1,204 @@
+        when (actionEvent) {
+            is BlazeActionEvent.FinishActivity -> requireActivity().finish()
+            is BlazeActionEvent.LaunchExternalBrowser -> {
+                ActivityLauncher.openUrlExternal(
+                    requireContext(),
+                    actionEvent.url
+                )
+            }
+        }
+    }
+
+    // The next 2 Composable(s) live in the fragment because they need access to the chromeClient outside of the fun
+    // Also the clients needs access to activity and we are not holding on to that elsewhere
+    @Composable
+    private fun BlazeWebViewScreen(
+        viewModel: BlazeWebViewViewModel = androidx.lifecycle.viewmodel.compose.viewModel()
+    ) {
+        val data by viewModel.model.collectAsState()
+        val blazeHeaderState by viewModel.blazeHeaderState.collectAsState()
+        Scaffold(
+            topBar = { TopAppBar(blazeHeaderState) },
+            content = { BlazeWebViewContent(data) }
+        )
+    }
+
+    @SuppressLint("SetJavaScriptEnabled")
+    @Composable
+    private fun BlazeWebViewContent(model: BlazeWebViewContentUiState) {
+        AndroidView(factory = {
+            WebView(it).apply {
+                layoutParams = ViewGroup.LayoutParams(
+                    ViewGroup.LayoutParams.MATCH_PARENT,
+                    ViewGroup.LayoutParams.MATCH_PARENT
+                )
+                scrollBarStyle = View.SCROLLBARS_INSIDE_OVERLAY
+                settings.userAgentString = model.userAgent
+                settings.javaScriptEnabled = model.enableJavascript
+                settings.domStorageEnabled = model.enableDomStorage
+                webViewClient = BlazeWebViewClient(this@BlazeWebViewFragment)
+                chromeClient = WPWebChromeClientWithFileChooser(
+                    activity,
+                    this,
+                    R.drawable.media_movieclip,
+                    null,
+                    this@BlazeWebViewFragment
+                )
+                webChromeClient = chromeClient
+                postUrl(WPCOM_LOGIN_URL, model.addressToLoad.toByteArray())
+            }
+        })
+    }
+
+    @Composable
+    private fun TopAppBar(
+        state: BlazeWebViewHeaderUiState
+    ) {
+        TopAppBar(
+            backgroundColor = MaterialTheme.colors.surface,
+            contentColor = MaterialTheme.colors.onSurface,
+            elevation = 0.dp,
+            title = withFullContentAlpha {
+                Text(stringResource(id = state.headerTitle))
+            },
+            actions = {
+                TextButton(
+                    onClick = {
+                        when (state) {
+                            is BlazeWebViewHeaderUiState.DoneAction ->
+                                blazeWebViewViewModel.onHeaderDoneActionClick()
+                            is BlazeWebViewHeaderUiState.EnabledCancelAction ->
+                                blazeWebViewViewModel.onHeaderCancelActionClick()
```

Comment:
💡 Can we move this `when`check to the ViewModel. I think exposing a click function in ViewModel and then checking which state in `BlazeWebViewViewModel` would be better. The rule I follow: the lesser the functions exposed from ViewModel, the better, also less logic in UI.

Reply:
Love this idea, thanks for the suggestion. ❤️  Ideally the click handler should be in the state, but I couldn't get it working properly, so this will do for now. I will revisit in the future when I have more time.
In commit [b79fa80](https://github.com/wordpress-mobile/WordPress-Android/pull/18013/commits/b79fa80220b88a4842794bafede5a4a32aefe075)